dags/multipod/maxtext_end_to_end.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A DAG to run end-to-end MaxText tests."""
+
+
+import datetime
+from airflow import models
+from dags import composer_env
+from dags.vm_resource import TpuVersion, Zone
+from dags.multipod.configs import maxtext_gce_config
+from dags.multipod.configs.common import SetupMode
+
+
+# Run once a day at 4 am UTC (8 pm PST)
+SCHEDULED_TIME = "0 4 * * *" if composer_env.is_prod_env() else None
+
+
+with models.DAG(
+    dag_id="maxtext_end_to_end",
+    schedule=SCHEDULED_TIME,
+    tags=["multipod_team", "maxtext", "stable", "nightly"],
+    start_date=datetime.datetime(2024, 1, 19),
+    catchup=False,
+) as dag:
+  test_name_prefix = "maxtext"
+
+  maxtext_stable_llama2_v4_8 = maxtext_gce_config.get_maxtext_end_to_end_test_config(
```

Comment:
This is fine but it seems duplicative -- I think all tests should be automatically set up to run on both stable and nightly.

Reply:
I see. If that is the case, we could have a for loop make it more concise. 

The benefit for this implementation is we can define any run order of tests (if needed), i.e. `test 1-stable >> test 2-stable >> test 3-stable` or `test1-stable >> test1-nightly`.

cc @jonb377

Reply:
@rwitten, will this for loop change make it more concise? Let me know your thoughts!